src/clients/GithubClient.ts

```diff
@@ -0,0 +1,282 @@
+}
+export interface ContentFileLinks {
+  git: string | null;
+  html: string | null;
+  self: string;
+}
+
+export interface ReactionRollup {
+  url: string;
+  total_count: number;
+  "+1": number;
+  "-1": number;
+  laugh: number;
+  confused: number;
+  heart: number;
+  hooray: number;
+  eyes: number;
+  rocket: number;
+}
+
+export interface GitHubErrorResponse {
+  message: string;
+  documentation_url?: string;
+  status?: string;
+  error?: string;
+}
+
+/**
+ * Client configuration options
+ */
+export interface GitHubClientOptions {
+  token?: string;
+  baseUrl?: string;
+}
+
+/**
+ * GitHub API client interface
+ */
+export interface GitHubApiClient {
+  /**
+   * Send a generic request to any GitHub API endpoint
+   * @param endpoint The API endpoint path (starting with /)
+   * @param params Optional query parameters
+   * @returns Promise with the API response
+   */
+  request<T>(
+    endpoint: string,
+    params?: Record<string, string | number | boolean>,
+  ): Promise<T>;
+
+  /**
+   * Get the latest release for a repository
+   * @param user GitHub username or organization
+   * @param repo Repository name
+   * @returns Promise with the latest release data
+   */
+  getLatestRelease(user: string, repo: string): Promise<LatestRelease>;
+
+  /**
+   * Get all releases for a repository
+   * @param user GitHub username or organization
+   * @param repo Repository name
+   * @returns Promise with an array of releases
+   */
+  getReleases(user: string, repo: string): Promise<Release[]>;
+
+  /**
+   * Get the README for a repository
+   * @param user GitHub username or organization
+   * @param repo Repository name
+   * @returns Promise with the README
+   * */
+  getReadme(user: string, repo: string): Promise<String>;
+}
+
+export interface Asset {
+  name: string;
+  downloadCount: number;
+  size: number;
+  browserDownloadUrl: string;
+}
+
+export interface LatestRelease {
+  id: number;
+  htmlUrl: string;
+  tagName: string;
+  publishedAt: string | null;
+  description: string | null;
+  sourceZip: string | null;
+  sourceTar: string | null;
+  assets: Asset[];
+}
+
+export interface Release {
+  id: number;
+  tagName: string;
+  publishedAt: string | null;
+}
+
+export class GitHubClient implements GitHubApiClient {
+  private baseUrl: string;
+  private headers: Record<string, string>;
+
+  constructor(options: GitHubClientOptions = {}) {
+    this.baseUrl = options.baseUrl || "https://api.github.com";
+    this.headers = {
+      Accept: "application/vnd.github+json",
+      "User-Agent": "astro-github-client",
+    };
+
+    if (options.token) {
+      this.headers["Authorization"] = `token ${options.token}`;
+    }
+  }
+
+  async request<T>(
+    endpoint: string,
+    params: Record<string, string | number | boolean> = {},
+  ): Promise<T> {
+    const url = new URL(`${this.baseUrl}${endpoint}`);
+
+    Object.entries(params).forEach(([key, value]) => {
+      if (value !== undefined && value !== null) {
```

Comment:
```suggestion
      if (!value) {
```

!value łapie też wartość "", ale w tym przypadku pusty query param tez można wywalić